### DISCUSSION TITLE
Problems with PM_compare()

### DISCUSSION THREAD

**Discussion by mat194** [LAPKB/Pmetrics#226] [Q&A]
Hi everyone! I'm starting to delve into popPK modeling, and I'm encountering an issue with the PM_compare() command in Pmetrics. I keep getting the error "arguments imply differing number of rows: x, y," even when I try to compare a model to itself. So far, I've created very simple models using both parametric and nonparametric fits, and I've been retrieving information manually from run$cycle$aic, run$cycle$bic, and run$cycle$ll.

Interestingly, this error doesn't occur when using the example file. Has anyone else experienced this issue or have any suggestions on how to resolve it?

**Comment by mnneely** [comment]
If there is a way you can post the PMout.Rdata file(s) here, we can test. It’s certainly possible there is a bug, and even if not, there should be some more informative error reporting. Those file(s) will be in the /outputs folder of the run(s).

**Reply by mat194** [reply]
I uploaded them in my Github: https://github.com/mat194/PK-PD.git

**Reply by mnneely** [reply]
Hmm, what version of Pmetrics are you using? Did you download from the master branch, the dev branch or another. The PMout.Rdata objects are malformed in that they have both an $NPdata and $ITdata field. There should only be one or the other. Based on the $final field, it looks like you did IT2B runs. Is that true?

**Reply by mat194** [reply]
I'm using version 2.1.1, which I believe I downloaded from the master branch. I tried deleting and redownloading the package using devtools::install_github("LAPKB/Pmetrics"), but the PM_compare function still doesn't work. The runs in question were IT2B. The PM_compare function works when comparing NPAG runs but fails when comparing two IT2B runs or one IT2B run with one NPAG run.

**Reply by mnneely** [reply]
Helpful thanks. Clearly there's a bug with the IT2B portion, which I admit gets much less love from us these days than NPAG. I'll get a fix pushed out shortly.
